module/dal/control/src/main/java/org/openbase/bco/dal/control/action/ActionImpl.java

```diff
@@ -728,6 +738,14 @@ private void cancelActionTask() {
         // cancel if exist
         if (actionTask != null) {
             actionTask.cancel(true);
+
+            // in case the task is directly cancel by this because the task was never canceled, we need to notify
+            // all waiting tasks about the cancellation
+            synchronized (actionTaskLock) {
+                if (isActionTaskFinished()) {
```

Comment:
I think the code would be more expressive if we would just check the `actionTaskRunning` flag here.